bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/handler/TriggerHandlerCallback.java

```diff
@@ -49,7 +48,7 @@
      *            <li><code>value</code> - represents output value of the {@link Trigger}'s {@link Output}
      *            </ul>
      */
-    public void triggered(Trigger trigger, @Nullable Map<String, ?> context);
+    public void triggered(Trigger trigger, Map<String, ?> context);
```

Comment:
@wborn I am in general a fan of the [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle), hence I prefer to have such method inputs being treated conservatively - i.e. also allowing null values as it is here seems to be an optional parameter anyway.
Why do you think it should be made a non-null param?

Reply:
I approached it from a clean code standpoint where allowing null values results in more complex code (potential bugs) and more unneccessary null checks, see https://github.com/openhab/openhab-core/pull/2763#discussion_r806764518.

This page summarizes what's in the Clean Code book on this subject: https://blog.markpearl.co.za/Working-With-Nulls

If providing no context is common, we could also add a `public void triggered(Trigger trigger)` method that calls `triggered(trigger, Map.of())`

Reply:
> If providing no context is common, we could also add a public void triggered(Trigger trigger) method that calls triggered(trigger, Map.of())

I am not sure how common it is, but it is possible. So yes, instead of forcing the consumer to add a `Map.of()` as a second parameter, offering the method with a single parameter is probably the cleanest option.

Reply:
Thanks!